File: music-web/src/api/playback/player-state.js
import client from '../client.js'

/**
 * music-playback 模块 — sync API。
 *
 * 后端对应：
 * - com.gjl.music.playback.sync.PlayerStateController — 播放状态同步（多端协同）
 */

/** 上报播放状态（客户端定时发送） */
export function reportPlayerState(state) {
  return client.post('/player/state', state)
}

/**
 * 获取播放状态。
 * @param {string} [username] — 不传则获取当前用户状态，传则获取指定用户状态
 */
export function getPlayerState(username) {
  if (username) {
    return client.get(`/player/state/${username}`)
  }
  return client.get('/player/state')
}

/** 发送播放控制事件（play/pause/stop/seek） */
export function sendControlEvent(event, songId, position) {
  return client.post('/player/state/control', { event, songId, position })
}

File: music-playback/src/main/java/com/gjl/music/playback/model/PlayerState.java
package com.gjl.music.playback.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 播放器状态快照。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PlayerState {
    private Long currentSongId;
    private String username;
    private double position;     // 当前播放位置（秒）
    private Long timestamp;      // 状态时间戳（毫秒）
    private String event;        // PLAY / PAUSE / STOP / TRACK_CHANGE / SEEK
    private int queueIndex;      // 队列中的位置（-1 表示不在队列中）
    private String queueId;      // 当前队列 ID
}
